packages/core/utils/src/dml/helpers/entity-builder/build-indexes.ts

```diff
@@ -42,3 +42,33 @@ function transformWhere(where?: string): string {
 
   return where
 }
+
+/*
+  When a "belongsTo" relationship is found on the DML schema, we create a default index on the
+  foreign key name of the field.
+*/
+export function buildForeignKeyIndexes<TSchema extends DMLSchema>(
```

Comment:
todoL: I believe the foreign key indexes should be managed when we transform to mikro orm entity, in this area, but I think we can prevent it from happening at the DML level where the foreign key does not yet really exists.

Reply:
My thinking is that the way foreign keys are managed will depends on the adapter that will be used. In our mikro orm util, when we create the relation we also create the foreign key complement and this is where I would probably see the index being added as we discussed

Reply:
good point. Will move it to mikroorm layer.